### PR TITLE
Clear padding bit when checking for overlaps

### PR DESCRIPTION
**Pull request by mpetrunic**
@wemeetagain Could you check if I'm creating proper mask?

### CODE REVIEW

File: src/base.ts [hunk 1]
```diff
@@ -74,8 +74,14 @@ export abstract class BitArray {
 
     for(let i = 0; i < this.byteArray.length; i++) {
       //invert byte from other array and xor against original, then xor against other array
+      //applying this mask won't change anything
+      let maskLengthBit = 255;
+      if(i === this.byteArray.length -1) {
```
Comment (wemeetagain) [inline src/base.ts]:
```suggestion
      if(i === this.byteArray.length - 1) {
```

File: src/base.ts [hunk 2]
```diff
@@ -74,8 +74,14 @@ export abstract class BitArray {
 
     for(let i = 0; i < this.byteArray.length; i++) {
       //invert byte from other array and xor against original, then xor against other array
+      //applying this mask won't change anything
```
[the remark this answers is not in the document]
Comment (mpetrunic) [reply src/base.ts]: yeah, this comment is meant that 255 mask applied won't change anything

**Comment by mpetrunic** [comment]
> The mask lgtm.
> One thing that's kinda strange is that this is bitlist specific logic going in src/base instead of src/list

because bitvector doesn't have padding bit? Can this affect overlap in bitvector?

not sure how to do it then without copy pasting a lot of complex logic in both of them

**Comment by wemeetagain** [comment]
This code won't effect bitvector.

Not really sure how to do it w/o copy pasta either, more just pointing it out :woman_shrugging:

**Pull request merged**